omnigibson/prims/xform_prim.py

```diff
@@ -155,13 +144,30 @@ def _set_xform_properties(self):
             f"old_orn: {current_orientation}, new_orn: {new_orientation}"
         )
 
+    @property
+    def _collision_filter_api(self):
+        return (
+            lazy.pxr.UsdPhysics.FilteredPairsAPI(self._prim)
+            if self._prim.HasAPI(lazy.pxr.UsdPhysics.FilteredPairsAPI)
+            else lazy.pxr.UsdPhysics.FilteredPairsAPI.Apply(self._prim)
+        )
+
+    @property
+    def _binding_api(self):
+        # TODO: Do we always need to apply this?
+        return (
+            lazy.pxr.UsdShade.MaterialBindingAPI(self.prim)
+            if self._prim.HasAPI(lazy.pxr.UsdShade.MaterialBindingAPI)
+            else lazy.pxr.UsdShade.MaterialBindingAPI.Apply(self.prim)
```

Comment:
nice -- this is done so that it's computed JIT, right? Does this cause any issues if it's not explicitly created at runtime?

Reply:
Yes! It seems to be working now, but I'll keep an eye on this